assets/src/components/checklist/index.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * Checklist for tracking multi-step tasks.
+ */
+
+/**
+ * WordPress dependencies.
+ */
+import { __ } from '@wordpress/i18n';
+import { Component, Children, cloneElement } from '@wordpress/element';
+import { Dashicon } from '@wordpress/components';
+
+/**
+ * Internal dependencies.
+ */
+import murielClassnames from '../../shared/js/muriel-classnames';
+import Button from '../button';
+import ProgressBar from '../progressBar';
+import './style.scss';
+
+class Checklist extends Component {
+
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			hideCompleted: false,
+		}
+	}
+
+	/**
+	 * Render.
+	 */
+	render() {
+		const { className, children, currentTask, progressBarText, ...otherProps } = this.props;
+		const { hideCompleted } = this.state;
+		const completedLabel = hideCompleted ? __( 'Show completed' ) : __( 'Hide completed' );
+		const completedIcon = hideCompleted ? 'arrow-down-alt2' : 'arrow-up-alt2';
+		const classes = murielClassnames(
+			'muriel-checklist',
+			className,
+			hideCompleted && 'is-hide-completed'
+		);
+		const shouldShowHideCompletionUI = currentTask > 0;
+		return (
+			<div className={ classes } { ...otherProps }>
+				<div className="muriel-checklist__header">
+					<div className="muriel-checklist__header-main">
+						<ProgressBar completed={ currentTask } total={ Children.count( children ) } displayFraction label={ progressBarText } />
+					</div>
+					{ shouldShowHideCompletionUI && (
+						<div className="muriel-checklist__header-secondary">
+							<label for="muriel-checklist__header-action">{ completedLabel }</label>
```

Comment:
There's a small warning when the "Do it" button is clicked

![image](https://user-images.githubusercontent.com/29167323/57785712-8e3ddd80-7732-11e9-9a62-ef148568a8e9.png)

And if we just update the `from` to `htmlFrom` in this line here, it goes away:
```suggestion
							<label htmlFor="muriel-checklist__header-action">{ completedLabel }</label>							
```

Reply:
Nice catch! Fixed in https://github.com/Automattic/newspack-plugin/pull/58/commits/b05f14cbad553c5fb8cd24f76905a039959878b8